api/v1alpha1/monitoringconfig_types.go

```diff
@@ -0,0 +1,75 @@
+// everest-operator
+// Copyright (C) 2022 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MonitoringType is a type of monitoring.
+type MonitoringType string
+
+const (
+	// PMM represents monitoring via PMM.
+	PMM MonitoringType = "pmm"
+)
+
+// MonitoringConfigSpec defines the desired state of MonitoringConfig.
+type MonitoringConfigSpec struct {
+	// Type is type of monitoring.
+	// +kubebuilder:validation:Enum=pmm
+	Type MonitoringType `json:"type,omitempty"`
+	// CredentialsSecretName is the name of the secret with credentials.
+	CredentialsSecretName string `json:"credentialsSecretName,omitempty"`
+	// PMM is configuration for the PMM monitoring type.
+	PMM PMMConfig `json:"pmm,omitempty"`
+}
+
+// PMMConfig is configuration of the PMM monitoring type.
+type PMMConfig struct {
+	// URL is url to the monitoring instance.
+	URL string `json:"url,omitempty"`
+	// Image is a Docker image name to use for deploying PMM client. Defaults to using the latest version.
+	Image string `json:"image,omitempty"`
```

Comment:
I actually think we might be able to go without this field.
I think we should query the PMM server for the version it’s running and use the appropriate image.
We can offload this to the user and force it to specify it (it would be great not having to though) but I don't think that using the `latest` tag (or any default tag) is a good idea.

Reply:
I agree it would be best to leave it out.
I don't think it's possible at the moment though, but shall be possible in the future.

At the moment we need to give the user the option to configure the version of `pmm-client` since we don't have the version detection.
The detection mechanism needs to regularly check what the version is and change the image name accordingly. I think it's a bigger topic and we may need to include the PMM team in the discussion to figure out how to approach this.

Reply:
I agree with you but in this case I think we should then make this field mandatory (remove the omitempty) because we can't guarantee that the `latest` client tag is compatible with the server version the user is running.

Reply:
Updated